Separate input moving to utils file

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #77187

In preparation for FSDP `device_id` support which will move GPU inputs to the device specified, move these codes to a utils file.

TODO: type the code: https://github.com/pytorch/pytorch/issues/77190

rohan-varma committed:
Separate input moving to utils file

[ghstack-poisoned]

vadimkantorov commented:
Related request on implementing a recursive utility of moving to target device into core `torch.to` function: https://github.com/pytorch/pytorch/issues/69431

awgu reviewed (approved):
LGTM!

awgu commented:
Maybe check the lint errors:
```
See https://www.flake8rules.com/rules/E306.html
2022-05-10T19:25:22.3542916Z ##[warning]'collections.abc' imported but unused
See https://www.flake8rules.com/rules/F401.html
2022-05-10T19:25:22.3544924Z ##[warning]'._functions._get_stream' imported but unused
```

rohan-varma committed:
Update on "Separate input moving to utils file"


In preparation for FSDP `device_id` support which will move GPU inputs to the device specified, move these codes to a utils file.

TODO: type the code: https://github.com/pytorch/pytorch/issues/77190

[ghstack-poisoned]

rohan-varma committed:
Update on "Separate input moving to utils file"


In preparation for FSDP `device_id` support which will move GPU inputs to the device specified, move these codes to a utils file.

TODO: type the code: https://github.com/pytorch/pytorch/issues/77190

[ghstack-poisoned]

rohan-varma committed:
Update on "Separate input moving to utils file"


In preparation for FSDP `device_id` support which will move GPU inputs to the device specified, move these codes to a utils file.

TODO: type the code: https://github.com/pytorch/pytorch/issues/77190

[ghstack-poisoned]

rohan-varma committed:
Update on "Separate input moving to utils file"


In preparation for FSDP `device_id` support which will move GPU inputs to the device specified, move these codes to a utils file.

TODO: type the code: https://github.com/pytorch/pytorch/issues/77190

[ghstack-poisoned]

rohan-varma committed:
Update on "Separate input moving to utils file"


In preparation for FSDP `device_id` support which will move GPU inputs to the device specified, move these codes to a utils file.

TODO: type the code: https://github.com/pytorch/pytorch/issues/77190

[ghstack-poisoned]

rohan-varma commented on torch/nn/parallel/distributed.py:
```diff
@@ -36,9 +36,8 @@
 from ..modules import Module
-from ._functions import _get_stream
 from ._replicated_tensor_ddp_utils import _ddp_with_replicated_tensor_enabled
-from .scatter_gather import gather, is_namedtuple, scatter_kwargs
```
actually we should not remove is_namedtuple due to bc

rohan-varma committed:
Update on "Separate input moving to utils file"


In preparation for FSDP `device_id` support which will move GPU inputs to the device specified, move these codes to a utils file.

TODO: type the code: https://github.com/pytorch/pytorch/issues/77190

[ghstack-poisoned]

rohan-varma commented:
@pytorchbot merge this please
